lib/icinga/apiactions.cpp

```diff
@@ -683,6 +681,25 @@ Dictionary::Ptr ApiActions::ExecuteCommand(const ConfigObject::Ptr& object,
 	MessageOrigin::Ptr origin = new MessageOrigin();
 	listener->RelayMessage(origin, checkable, updateMessage, true);
 
+	double scheduled_start = checkable->GetNextCheck();
+	double before_check = Utility::GetTime();
+
+	CheckResult::Ptr cr = new CheckResult();
+	cr->SetScheduleStart(scheduled_start);
+	cr->SetExecutionStart(before_check);
+
+	Dictionary::Ptr execMacros = new Dictionary();
+	if (command_type == "CheckCommand") {
+		CheckCommand::Ptr cmd = CheckCommand::GetByName(resolved_command);
+		cmd->Execute(checkable, cr, execMacros, false);
+	} else if (command_type == "EventCommand") {
+		EventCommand::Ptr cmd = EventCommand::GetByName(resolved_command);
+		cmd->Execute(checkable, execMacros, false);
+	} else if (command_type == "NotificationCommand") {
+		/* TODO */
+		return ApiActions::CreateResult(501, "Not implementd.");
```

Comment:
notification & user: add these two as action HTTP parameters (default: "") and resolve+GetByName() them if `command_type == "NotificationCommand"`
cr => checkable's last CR
type => custom
author => the user calling the action
comment => ""